core-data/entities.js

```diff
@@ -1,23 +1,42 @@
 /**
  * External dependencies
  */
-import { find, upperFirst, camelCase } from 'lodash';
+import { upperFirst, camelCase, map, find } from 'lodash';
 
-const entities = [
+/**
+ * WordPress dependencies
+ */
+import apiRequest from '@wordpress/api-request';
+
+/**
+ * Internal dependencies
+ */
+import { getEntitiesByKind } from './selectors';
+import { addEntities } from './actions';
+
+export const defaultEntities = [
 	{ name: 'postType', kind: 'root', key: 'slug', baseUrl: '/wp/v2/types' },
 	{ name: 'media', kind: 'root', baseUrl: '/wp/v2/media' },
 ];
 
+export const kinds = [
+	{ name: 'postType', loadEntities: loadPostTypeEntities },
+];
+
 /**
- * Returns the entity object given its kind and name.
- *
- * @param {string} kind  Entity kind.
- * @param {string} name  Entity name.
+ * Returns the list of post type entities.
  *
- * @return {Object} Entity
+ * @return {Array} entities
```

Comment:
Pedantry: An `async` function always returns type `Promise`†. In this case it's a promise which resolves with an array.

_† ...which ESLint makes quite annoying for its `valid-jsdoc` rule since it assumes the presence of a `return` statement, which isn't always the case:_

```js
async function sleep( duration ) {
	return new Promise( ( resolve ) => {
		setTimeout( resolve, duration );
	} );
}

async function resolve() {
	await sleep( 1000 );
	await sleep( 1000 );
}

console.log( resolve() instanceof Promise );
// true
```